File: src/test/java/io/binac/leetcode/GasStationTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class GasStationTest {
    private final GasStation.Solution1 solution1 = new GasStation.Solution1();

    @Test
    void test1() {
        int gas1[] = {1, 2, 3, 4, 5}, cost1[] = {3, 4, 5, 1, 2};
        assertEquals(3, solution1.canCompleteCircuit(gas1, cost1));

        int gas2[] = {2, 3, 4}, cost2[] = {3, 4, 3};
        assertEquals(-1, solution1.canCompleteCircuit(gas2, cost2));
    }
}

File: cpp/tests/GasStationTest.cpp
#include "GasStation.h"
#include "gtest/gtest.h"
#include <string>

using namespace lcpp;

TEST(GasStation, Solution134_1) {
  auto S134_1 = Solution134_1();

  std::vector<int> Gas1{1, 2, 3, 4, 5}, Cost1{3, 4, 5, 1, 2};
  int const Expected1 = 3;
  EXPECT_EQ(Expected1, S134_1.canCompleteCircuit(Gas1, Cost1));

  std::vector<int> Gas2{2, 3, 4}, Cost2{3, 4, 3};
  int const Expected2 = -1;
  EXPECT_EQ(Expected2, S134_1.canCompleteCircuit(Gas2, Cost2));
}